shopinvader_api_cart/routers/cart.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from collections import OrderedDict
+from typing import Annotated
+
+from fastapi import APIRouter, Depends
+
+from odoo import _, api, models
+from odoo.exceptions import MissingError
+from odoo.tools import float_compare
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+)
+from odoo.addons.sale.models.sale_order import SaleOrder
+from odoo.addons.sale.models.sale_order_line import SaleOrderLine
+
+from ..schemas import CartResponse, CartSyncInput, CartTransaction
+
+cart_router = APIRouter()
+
+
+@cart_router.get("/", response_model=CartResponse | dict)
+@cart_router.get("/{uuid}", response_model=CartResponse | dict)
+def get(
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated["ResPartner", Depends(authenticated_partner)],
+    uuid: str | None = None,
+) -> CartResponse:
+    """
+    Return an empty dict if no cart was found
+    """
+    cart = env["sale.order"]._find_open_cart(partner.id, uuid)
+    return CartResponse.from_cart(cart) if cart else {}
+
+
+@cart_router.post("/sync", response_model=CartResponse | dict, status_code=201)
+@cart_router.post("/sync/{uuid}", response_model=CartResponse | dict, status_code=201)
+def sync(
+    data: CartSyncInput,
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated["ResPartner", Depends(authenticated_partner)],
+    uuid: str | None = None,
+) -> CartResponse:
+    cart = env["sale.order"]._find_open_cart(partner.id, uuid)
+    cart = env["shopinvader_api_cart.manager"]._sync_cart(
+        partner, cart, uuid, data.transactions
+    )
+    return CartResponse.from_cart(cart) if cart else {}
+
+
+class ShopinvaderApiCartManager(models.AbstractModel):
+    _name = "shopinvader_api_cart.manager"
```

Comment:
I'm not a big fan of this name but I don't have a better idea for now...

Reply:
What about?
```suggestion
    _name = "shopinvader_api_cart.service.helper"
```

Reply:
Done